command/agent/cert_end_to_end_test.go

```diff
@@ -580,4 +543,9 @@ func TestCertEndToEnd_CertsInConfig(t *testing.T) {
 
 		time.Sleep(250 * time.Millisecond)
 	}
+
+	err = <-errchan
```

Comment:
The tests are passing, but for the wrong reasons. The tests are getting context canceled after their 30s timeout when ran locally (taking ~33s per case), but we don't report this as a test failure. The same tests on the base branch takes about 4s per test case.